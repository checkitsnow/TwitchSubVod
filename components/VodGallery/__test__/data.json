{
  "_total": 2,
  "videos": [
    {
      "title": "[ NoPixel ] FROM 20K TO 100K OR BUST. BEST PLAYER WORLD OF ALL GAMES, BIG BIG.",
      "description": null,
      "description_html": null,
      "broadcast_id": 41192454541,
      "broadcast_type": "archive",
      "status": "recorded",
      "tag_list": "",
      "views": 3229108,
      "url": "https://www.twitch.tv/videos/965866667",
      "language": "en",
      "created_at": "2021-03-28T01:00:41Z",
      "viewable": "public",
      "viewable_at": null,
      "published_at": "2021-03-27",
      "delete_at": "2021-05-27T01:00:41Z",
      "_id": "v965866667",
      "recorded_at": "2021-03-28",
      "game": "Just Chatting",
      "length": 63043,
      "preview": {
        "small": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/bb5ee1fa51385b4119f4_xqcow_41192454541_1616893232//thumb/thumb0-80x45.jpg",
        "medium": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/bb5ee1fa51385b4119f4_xqcow_41192454541_1616893232//thumb/thumb0-320x180.jpg",
        "large": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/bb5ee1fa51385b4119f4_xqcow_41192454541_1616893232//thumb/thumb0-640x360.jpg",
        "template": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/bb5ee1fa51385b4119f4_xqcow_41192454541_1616893232//thumb/thumb0-{width}x{height}.jpg"
      },
      "animated_preview_url": "https://d2nvs31859zcd8.cloudfront.net/bb5ee1fa51385b4119f4_xqcow_41192454541_1616893232/storyboards/965866667-strip-0.jpg",
      "thumbnails": {
        "small": [
          {
            "type": "generated",
            "url": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/bb5ee1fa51385b4119f4_xqcow_41192454541_1616893232//thumb/thumb0-80x45.jpg"
          },
          {
            "type": "generated",
            "url": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/bb5ee1fa51385b4119f4_xqcow_41192454541_1616893232//thumb/thumb1-80x45.jpg"
          },
          {
            "type": "generated",
            "url": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/bb5ee1fa51385b4119f4_xqcow_41192454541_1616893232//thumb/thumb2-80x45.jpg"
          },
          {
            "type": "generated",
            "url": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/bb5ee1fa51385b4119f4_xqcow_41192454541_1616893232//thumb/thumb3-80x45.jpg"
          }
        ],
        "medium": [
          {
            "type": "generated",
            "url": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/bb5ee1fa51385b4119f4_xqcow_41192454541_1616893232//thumb/thumb0-320x180.jpg"
          },
          {
            "type": "generated",
            "url": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/bb5ee1fa51385b4119f4_xqcow_41192454541_1616893232//thumb/thumb1-320x180.jpg"
          },
          {
            "type": "generated",
            "url": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/bb5ee1fa51385b4119f4_xqcow_41192454541_1616893232//thumb/thumb2-320x180.jpg"
          },
          {
            "type": "generated",
            "url": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/bb5ee1fa51385b4119f4_xqcow_41192454541_1616893232//thumb/thumb3-320x180.jpg"
          }
        ],
        "large": [
          {
            "type": "generated",
            "url": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/bb5ee1fa51385b4119f4_xqcow_41192454541_1616893232//thumb/thumb0-640x360.jpg"
          },
          {
            "type": "generated",
            "url": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/bb5ee1fa51385b4119f4_xqcow_41192454541_1616893232//thumb/thumb1-640x360.jpg"
          },
          {
            "type": "generated",
            "url": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/bb5ee1fa51385b4119f4_xqcow_41192454541_1616893232//thumb/thumb2-640x360.jpg"
          },
          {
            "type": "generated",
            "url": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/bb5ee1fa51385b4119f4_xqcow_41192454541_1616893232//thumb/thumb3-640x360.jpg"
          }
        ],
        "template": [
          {
            "type": "generated",
            "url": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/bb5ee1fa51385b4119f4_xqcow_41192454541_1616893232//thumb/thumb0-{width}x{height}.jpg"
          },
          {
            "type": "generated",
            "url": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/bb5ee1fa51385b4119f4_xqcow_41192454541_1616893232//thumb/thumb1-{width}x{height}.jpg"
          },
          {
            "type": "generated",
            "url": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/bb5ee1fa51385b4119f4_xqcow_41192454541_1616893232//thumb/thumb2-{width}x{height}.jpg"
          },
          {
            "type": "generated",
            "url": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/bb5ee1fa51385b4119f4_xqcow_41192454541_1616893232//thumb/thumb3-{width}x{height}.jpg"
          }
        ]
      },
      "fps": {
        "160p30": 29.999545552998157,
        "360p30": 29.999545552998157,
        "480p30": 29.999545552998157,
        "720p30": 29.999545552998157,
        "720p60": 59.99907524397182,
        "chunked": 59.99907524397182
      },
      "seek_previews_url": "https://d2nvs31859zcd8.cloudfront.net/bb5ee1fa51385b4119f4_xqcow_41192454541_1616893232/storyboards/965866667-info.json",
      "resolutions": {
        "160p30": "284x160",
        "360p30": "640x360",
        "480p30": "852x480",
        "720p30": "1280x720",
        "720p60": "1280x720",
        "chunked": "1920x1080"
      },
      "restriction": "",
      "channel": {
        "mature": false,
        "status": "[ NoPixel ] X (JUST X) FIRST TRY SOON TO BE BACK TO BACK CERTIFIED BANK ROBBER EZ ðŸ”¥",
        "broadcaster_language": "en",
        "broadcaster_software": "unknown_rtmp",
        "display_name": "xQcOW",
        "game": "Grand Theft Auto V",
        "language": "en",
        "_id": 71092938,
        "name": "xqcow",
        "created_at": "2014-09-12T23:50:05.989719Z",
        "updated_at": "2021-03-28T21:50:23.054953Z",
        "partner": true,
        "logo": "https://static-cdn.jtvnw.net/jtv_user_pictures/xqcow-profile_image-9298dca608632101-300x300.jpeg",
        "video_banner": "https://static-cdn.jtvnw.net/jtv_user_pictures/cd4e260a-7ea8-476c-af64-a151838add9a-channel_offline_image-1920x1080.png",
        "profile_banner": "https://static-cdn.jtvnw.net/jtv_user_pictures/83e86af1-9a6c-42b1-98e2-3f6238a744b5-profile_banner-480.png",
        "profile_banner_background_color": "",
        "url": "https://www.twitch.tv/xqcow",
        "views": 306005067,
        "followers": 7970446,
        "broadcaster_type": "partner",
        "description": "Overwatch Professional tank player and full time streamer. EZ Clap",
        "private_video": false,
        "privacy_options_enabled": false
      },
      "increment_view_count_url": "https://countess.twitch.tv/ping.gif?u=%7B%22id%22%3A%22965866667%22%2C%22type%22%3A%22vod%22%7D"
    },
    {
      "title": "[ NoPixel ] BIG DAY TODAY OMG! COURT CASE, CAZINO JUMP AND MORE WOW! WARLORD MOMENT INBOUND",
      "description": null,
      "description_html": null,
      "broadcast_id": 42115294974,
      "broadcast_type": "archive",
      "status": "recorded",
      "tag_list": "",
      "views": 2876532,
      "url": "https://www.twitch.tv/videos/964333108",
      "language": "en",
      "created_at": "2021-03-26T22:40:17Z",
      "viewable": "public",
      "viewable_at": null,
      "published_at": "2021-03-25",
      "delete_at": "2021-05-25T22:40:17Z",
      "_id": "v964333108",
      "recorded_at": "2021-03-26",
      "game": "Just Chatting",
      "length": 54516,
      "preview": {
        "small": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/cb1ba5495bbda35fd7ad_xqcow_42115294974_1616798408//thumb/thumb0-80x45.jpg",
        "medium": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/cb1ba5495bbda35fd7ad_xqcow_42115294974_1616798408//thumb/thumb0-320x180.jpg",
        "large": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/cb1ba5495bbda35fd7ad_xqcow_42115294974_1616798408//thumb/thumb0-640x360.jpg",
        "template": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/cb1ba5495bbda35fd7ad_xqcow_42115294974_1616798408//thumb/thumb0-{width}x{height}.jpg"
      },
      "animated_preview_url": "https://d2nvs31859zcd8.cloudfront.net/cb1ba5495bbda35fd7ad_xqcow_42115294974_1616798408/storyboards/964333108-strip-0.jpg",
      "thumbnails": {
        "small": [
          {
            "type": "generated",
            "url": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/cb1ba5495bbda35fd7ad_xqcow_42115294974_1616798408//thumb/thumb0-80x45.jpg"
          },
          {
            "type": "generated",
            "url": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/cb1ba5495bbda35fd7ad_xqcow_42115294974_1616798408//thumb/thumb1-80x45.jpg"
          },
          {
            "type": "generated",
            "url": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/cb1ba5495bbda35fd7ad_xqcow_42115294974_1616798408//thumb/thumb2-80x45.jpg"
          },
          {
            "type": "generated",
            "url": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/cb1ba5495bbda35fd7ad_xqcow_42115294974_1616798408//thumb/thumb3-80x45.jpg"
          }
        ],
        "medium": [
          {
            "type": "generated",
            "url": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/cb1ba5495bbda35fd7ad_xqcow_42115294974_1616798408//thumb/thumb0-320x180.jpg"
          },
          {
            "type": "generated",
            "url": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/cb1ba5495bbda35fd7ad_xqcow_42115294974_1616798408//thumb/thumb1-320x180.jpg"
          },
          {
            "type": "generated",
            "url": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/cb1ba5495bbda35fd7ad_xqcow_42115294974_1616798408//thumb/thumb2-320x180.jpg"
          },
          {
            "type": "generated",
            "url": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/cb1ba5495bbda35fd7ad_xqcow_42115294974_1616798408//thumb/thumb3-320x180.jpg"
          }
        ],
        "large": [
          {
            "type": "generated",
            "url": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/cb1ba5495bbda35fd7ad_xqcow_42115294974_1616798408//thumb/thumb0-640x360.jpg"
          },
          {
            "type": "generated",
            "url": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/cb1ba5495bbda35fd7ad_xqcow_42115294974_1616798408//thumb/thumb1-640x360.jpg"
          },
          {
            "type": "generated",
            "url": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/cb1ba5495bbda35fd7ad_xqcow_42115294974_1616798408//thumb/thumb2-640x360.jpg"
          },
          {
            "type": "generated",
            "url": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/cb1ba5495bbda35fd7ad_xqcow_42115294974_1616798408//thumb/thumb3-640x360.jpg"
          }
        ],
        "template": [
          {
            "type": "generated",
            "url": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/cb1ba5495bbda35fd7ad_xqcow_42115294974_1616798408//thumb/thumb0-{width}x{height}.jpg"
          },
          {
            "type": "generated",
            "url": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/cb1ba5495bbda35fd7ad_xqcow_42115294974_1616798408//thumb/thumb1-{width}x{height}.jpg"
          },
          {
            "type": "generated",
            "url": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/cb1ba5495bbda35fd7ad_xqcow_42115294974_1616798408//thumb/thumb2-{width}x{height}.jpg"
          },
          {
            "type": "generated",
            "url": "https://static-cdn.jtvnw.net/cf_vods/d2nvs31859zcd8/cb1ba5495bbda35fd7ad_xqcow_42115294974_1616798408//thumb/thumb3-{width}x{height}.jpg"
          }
        ]
      },
      "fps": {
        "160p30": 29.99945959908856,
        "360p30": 29.99945959908856,
        "480p30": 29.99945959908856,
        "720p60": 59.99891553086012,
        "chunked": 59.99706109950647
      },
      "seek_previews_url": "https://d2nvs31859zcd8.cloudfront.net/cb1ba5495bbda35fd7ad_xqcow_42115294974_1616798408/storyboards/964333108-info.json",
      "resolutions": {
        "160p30": "284x160",
        "360p30": "640x360",
        "480p30": "852x480",
        "720p60": "1280x720",
        "chunked": "1920x1080"
      },
      "restriction": "",
      "channel": {
        "mature": false,
        "status": "[ NoPixel ] X (JUST X) FIRST TRY SOON TO BE BACK TO BACK CERTIFIED BANK ROBBER EZ ðŸ”¥",
        "broadcaster_language": "en",
        "broadcaster_software": "unknown_rtmp",
        "display_name": "xQcOW",
        "game": "Grand Theft Auto V",
        "language": "en",
        "_id": 71092938,
        "name": "xqcow",
        "created_at": "2014-09-12T23:50:05.989719Z",
        "updated_at": "2021-03-28T21:50:23.054953Z",
        "partner": true,
        "logo": "https://static-cdn.jtvnw.net/jtv_user_pictures/xqcow-profile_image-9298dca608632101-300x300.jpeg",
        "video_banner": "https://static-cdn.jtvnw.net/jtv_user_pictures/cd4e260a-7ea8-476c-af64-a151838add9a-channel_offline_image-1920x1080.png",
        "profile_banner": "https://static-cdn.jtvnw.net/jtv_user_pictures/83e86af1-9a6c-42b1-98e2-3f6238a744b5-profile_banner-480.png",
        "profile_banner_background_color": "",
        "url": "https://www.twitch.tv/xqcow",
        "views": 306005067,
        "followers": 7970446,
        "broadcaster_type": "partner",
        "description": "Overwatch Professional tank player and full time streamer. EZ Clap",
        "private_video": false,
        "privacy_options_enabled": false
      },
      "increment_view_count_url": "https://countess.twitch.tv/ping.gif?u=%7B%22id%22%3A%22964333108%22%2C%22type%22%3A%22vod%22%7D"
    }
  ]
}
