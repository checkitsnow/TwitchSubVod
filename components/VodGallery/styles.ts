import styled, { css, keyframes } from 'styled-components';
import { appearFromTop } from '@/utils/animations/appearFromTop';

interface AnimatedProps {
  url: string;
  animated: any;
}

export const StreamerInformation = styled.div`
  * {
    margin: 0;
  }

  display: flex;
  margin-top: 3rem;
  max-width: 320px;
  color: var(--text);
  cursor: pointer;
  transition: all 0.2s ease;

  :hover {
    color: var(--purple);

    img {
      border: 2px solid var(--purple);
    }
  }

  div {
    img {
      width: 100px;
      height: 100px;
      border-radius: 50%;
      margin-right: 1rem;
      border: 2px solid var(--text);
    }

    display: flex;
    flex-direction: column;
    justify-content: space-evenly;

    p {
      overflow: hidden;
      text-overflow: ellipsis;
      display: -webkit-box;
      -webkit-line-clamp: 3;
      -webkit-box-orient: vertical;
    }
  }
`;

export const Container = styled.div`
  max-width: 60rem;
  display: flex;
  justify-content: center;
  flex-wrap: wrap;

  div {
    max-width: 19rem;
    margin: 0.5rem;
    display: flex;
    flex-direction: column;
    width: 19rem;

    button {
      width: 19rem;
      padding: 0;
      display: flex;
      flex-direction: column;
      background-color: transparent;
      color: var(--text);
      font-family: 'Roboto', sans-serif;

      &:hover {
        color: var(--purple);
      }

      div {
        max-height: 10.6875rem;
        margin: 0;

        &:hover {
          background: var(--purple);
        }
      }

      > span {
        display: flex;
        justify-content: space-between;
        width: 100%;
        p {
          margin: 0;
          font-size: 0.75rem;
          font-weight: normal;
        }
      }
    }

    strong {
      word-wrap: break-word;
      margin: 0.2rem 0;
      font-size: 0.9rem;
      font-weight: bold;
      max-width: 100%;
    }
  }

  div & {
    padding: 1rem 0;
  }
`;

const play = keyframes`
    100% {
        background-position: 0 -1800px;
      }
`;

export const Image = styled.figure<AnimatedProps>`
  ${(props) =>
    props.url &&
    css`
      background: url(${props.url}) no-repeat top center;
    `}
  background-size: contain;
  margin: 0;
  display: flex;
  height: 10.6875rem;
  width: 19rem;
  object-fit: contain;

  transition: transform 0.075s ease;
  animation: ${appearFromTop} 0.25s ease-out;

  &:hover {
    transform: translateX(0.5rem) translateY(-0.5rem);
    max-height: 10.6875rem;
    object-fit: cover;

    ${(props) =>
      props.animated
        ? css`
            background: url(${props.animated});
          `
        : css`
            background: url(${props.url});
          `}

    animation: ${play} 7s steps(10) infinite;
  }

  span {
    background: rgb(24, 26, 27, 0.6);
    color: #f1f1f1;
    position: absolute;
    margin: 0.5rem;
    padding: 0.15rem 0.2rem;
    border-radius: 0.1rem;
    font-weight: bold;
  }
`;
