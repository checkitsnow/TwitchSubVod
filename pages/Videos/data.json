{
  "mostSearchedStreamers": [
    "Cellbit",
    "Ricegum",
    "Rogersbase",
    "Berry0314",
    "Quesadacid",
    "Mizzkay",
    "Chapix",
    "Liberto",
    "Sozzlepopp",
    "Taylor_jevaux",
    "Insize",
    "Mira",
    "Yunicorn19",
    "Anas_off",
    "Shanasparklez",
    "Jinnytty",
    "Hermanli",
    "Asmrbby",
    "Snackless",
    "Silky",
    "Tarzaned",
    "Undefined",
    "Iwilldominate",
    "Adinross",
    "Nahamsec",
    "Creativemonkeyz",
    "Indiefoxx",
    "Yuggie_tv",
    "ZeratorAdinRoss",
    "Taylor_Jevaux",
    "Milliemiless",
    "Enotishka18",
    "Miamalkova",
    "Buckefps",
    "Cufffem",
    "DotaCapitalistJLTomy",
    "Kirio1984",
    "Emmalangevin",
    "CriticalroleEasterbedddIlonqueen",
    "Itsbinkybee",
    "SeredaJustyeseul",
    "Saudsqually",
    "Zathzen",
    "Ibabyrainbow",
    "Rob_lopezAircoolHowdoyoumakethat",
    "Kittycatnat1",
    "XixoBotKZ",
    "Domingo",
    "GothamChess",
    "Overcooler",
    "Sovitia",
    "AnthonyzChigua",
    "Demisux",
    "Gmhikaru",
    "Kamet0",
    "Taliyaahh",
    "Thathoneydip",
    "Annoying",
    "Athnessa",
    "Deejaypuffy",
    "Faiar",
    "Gavrilka",
    "LesstraMinxkittyy",
    "Peterhanstyle",
    "Rafilia",
    "Veronika",
    "Abu_abeer16",
    "Amouranth",
    "Avocadopeeled",
    "Imtiffany",
    "Kaileymae",
    "Mahdiba",
    "Mylandthailand",
    "Notanyla",
    "Tashaghouri",
    "CriticalRoleIvysky",
    "MARI",
    "Papomc",
    "Gothamchess",
    "Jasikyu_",
    "Jltomy",
    "Katthebratt",
    "Lirik",
    "Niva",
    "Novaruu",
    "Pink_sparkles",
    "Smqcked",
    "SOZZLEPOPP",
    "Supernick02",
    "Xoaeriel",
    "XoarielAileybeanssa",
    "Briannamarie_m",
    "BTBAMofficial",
    "Burnt_BiscuitytElded",
    "EvaannaGMHikaru",
    "HecatiazKrypto_nat",
    "Laurentracy92",
    "Marvincalifornia",
    "Miniminter",
    "Rita",
    "Saleenerz",
    "TheGamingDefinitionTV",
    "Tonytheduff",
    "Xqcow",
    "Yangile",
    "Yatesieboy"
  ]
}
