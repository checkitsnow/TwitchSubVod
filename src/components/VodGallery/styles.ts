import styled from 'styled-components';

export const Container = styled.div`
  max-width: 60rem;
  display: flex;
  justify-content: center;
  flex-wrap: wrap;

  div {
    max-width: 19rem;
    margin: 0.5rem;
    display: flex;
    flex-direction: column;

    a {
      display: flex;
      flex-direction: column;
      color: #efefef;

      &:hover {
        color: var(--purple);
      }

      div {
        margin: 0.5rem 0;
        background: var(--purple);
      }
    }

    img {
      max-width: 19rem;
      display: flex;
      object-fit: contain;
      transition: transform 0.075s ease;

      &:hover {
        transform: translateX(0.5rem) translateY(-0.5rem);
      }
    }

    strong {
      word-wrap: break-word;
    }

    p {
      margin: 0.1rem;
      font-size: 0.75rem;
    }
  }

  div & {
    padding: 1rem 0;
  }
`;
