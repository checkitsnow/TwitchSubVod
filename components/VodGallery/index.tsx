import React, { useEffect, useState } from 'react';
import axios from 'axios';
import ReactGA from 'react-ga';
import ErrorModal from '@/components/ErrorModal';
import { useGlobal } from '@/stores/GlobalContext';
import VodModal from '@/components/VodModal';

import { Container, Image } from './styles';

interface ResultProps {
  _id: string;
  animated_preview_url: string;
  preview: {
    large: string;
  };
  broadcast_id: number;
  title: string;
  views: number;
  url: string;
  length: number; //in seconds
  game: string;
  channel: {
    display_name: string;
  };
  recorded_at: string;
}

export const formatNumber = (num: number) => {
  if (num > 1000 && num < 1000000) {
    return (num / 1000).toString().slice(0, 3) + 'K';
  } else if (num > 1000000) {
    return (num / 1000000).toString().slice(0, 3) + 'M';
  } else {
    return num;
  }
};

const VodGallery = ({ data }: any) => {
  const { vodUrl, setVodUrl, videoQuality } = useGlobal();
  useEffect(() => {
    ReactGA.initialize(`${process.env.NEXT_PUBLIC_GOOGLE_TRACKING}`, {
      testMode: process.env.NODE_ENV === 'test',
    });
  }, []);

  const [error, setError] = useState(false);

  const handleVideo = async (result: any) => {
    const splitString = await result.animated_preview_url.split('/')[3];
    const hostUrl = await result.animated_preview_url.split('/')[2];
    let dataUrl = `${process.env.NEXT_PUBLIC_CORS}https://${hostUrl}/${splitString}/${videoQuality}/index-dvr.m3u8`;

    try {
      const checkIfVideoExists = await axios.head(`${dataUrl}`);
      if (checkIfVideoExists.status !== 403) {
        setVodUrl(result.broadcast_id !== 1 ? dataUrl : result.url);
      } else {
        setError(true);
      }
    } catch {
      setError(true);
    }

    window.scrollTo({ behavior: 'smooth', top: 300 });

    ReactGA.event({
      category: 'SearchedUserForDeletedVod',
      action: `${splitString}`,
    });
  };

  const timeToHMS = (s: number) => {
    const h = Math.floor(s / 3600);
    s -= h * 3600;
    const m = Math.floor(s / 60);
    s -= m * 60;

    return `${h}:${m > 10 ? m : '0' + m}:${s > 10 ? s : '0' + s}`;
  };

  return (
    <>
      {data && !error && <h1>Streamer: {data[0].channel.display_name}</h1>}
      {vodUrl && <VodModal videoUrl={vodUrl} />}
      {error && !vodUrl && videoQuality === 'chunked' && (
        <ErrorModal message="We couldn't find this video" />
      )}
      {error && !vodUrl && videoQuality !== 'chunked' && (
        <ErrorModal message="We couldn't find this video, try changing the quality to 'Source'" />
      )}

      <Container>
        {data.map((result: ResultProps) => {
          console.log(new Date(result.recorded_at).toLocaleDateString());
          return (
            <div key={result._id} title={result.title}>
              <button type="button" onClick={() => handleVideo(result)}>
                <div>
                  <Image
                    url={result.preview.large}
                    animated={result.animated_preview_url}
                  >
                    <span>{timeToHMS(result.length)}</span>
                  </Image>
                </div>
                <strong>{result.title}</strong>
                <span>
                  <p>Views: {formatNumber(result.views)}</p>
                  <p>{new Date(result.recorded_at).toLocaleDateString()}</p>
                </span>
              </button>
            </div>
          );
        })}
      </Container>
    </>
  );
};

export default VodGallery;
