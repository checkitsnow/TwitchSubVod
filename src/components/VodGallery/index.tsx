import axios from 'axios';
import React, { useEffect, useState } from 'react';
import ReactGA from 'react-ga';
import { FiAlertCircle } from 'react-icons/fi';

import VodModal from '../VodModal';

import { Error, Container, Image } from './styles';

interface ResultProps {
  _id: string;
  animated_preview_url: string;
  preview: {
    large: string;
  };
  broadcast_id: number;
  title: string;
  views: number;
  url: string;
  length: number; //in seconds
  game: string;
}

export const formatNumber = (num: number) => {
  if (num > 1000 && num < 1000000) {
    return (num / 1000).toString().slice(0, 3) + 'K';
  } else if (num > 1000000) {
    return (num / 1000000).toString().slice(0, 3) + 'M';
  } else {
    return num;
  }
};

const VodGallery = ({ data, quality }: any) => {
  useEffect(() => {
    ReactGA.initialize(`${process.env.REACT_APP_GOOGLE_TRACKING}`, {
      testMode: process.env.NODE_ENV === 'test',
    });
  }, []);

  const [vodUrl, setVodUrl] = useState(''); //use useContext later to clear this when we search again
  const [error, setError] = useState(false);

  const handleVideo = async (result: any) => {
    const splitString = result.preview.large.split('/')[5];
    let dataUrl = `https://twitch-cors.herokuapp.com/https://vod-secure.twitch.tv/${splitString}/${quality}/index-dvr.m3u8`;

    try {
      const checkIfVideoExists = await axios.head(`${dataUrl}`);
      console.log(checkIfVideoExists);
      if (checkIfVideoExists.status !== 403) {
        setVodUrl(result.broadcast_id !== 1 ? dataUrl : result.url);
      } else {
        setError(true);
      }
    } catch {
      setError(true);
    }

    window.scrollTo({ behavior: 'smooth', top: 300 });

    ReactGA.event({
      category: 'SearchedUserForDeletedVod',
      action: `${splitString}`,
    });
  };

  const timeToHMS = (s: number) => {
    const h = Math.floor(s / 3600);
    s -= h * 3600;
    const m = Math.floor(s / 60);
    s -= m * 60;

    return `${h}:${m > 10 ? m : '0' + m}:${s > 10 ? s : '0' + s}`;
  };

  return (
    <>
      {vodUrl && <VodModal videoUrl={vodUrl} />}
      {error && !vodUrl && (
        <Error>
          <FiAlertCircle size={36} />
          <span>Couldn't find this video</span>
        </Error>
      )}

      <Container>
        {data.map((result: ResultProps) => {
          return (
            <div key={result._id}>
              <button type="button" onClick={() => handleVideo(result)}>
                <div>
                  <Image
                    url={result.preview.large}
                    animated={result.animated_preview_url}
                  >
                    <span>{timeToHMS(result.length)}</span>
                  </Image>
                </div>
                <strong>{result.title}</strong>
                <p>Views: {formatNumber(result.views)}</p>
              </button>
            </div>
          );
        })}
      </Container>
    </>
  );
};

export default VodGallery;
