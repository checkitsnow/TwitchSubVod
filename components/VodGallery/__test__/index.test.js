import React from 'react';
import renderer from 'react-test-renderer';
import VodGallery from '..';
import data from './data.json';

it('should render Vod Gallery correctly', () => {
  const tree = renderer
    .create(<VodGallery data={data.videos} quality="chunked" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
