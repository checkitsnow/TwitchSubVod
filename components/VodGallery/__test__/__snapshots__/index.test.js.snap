// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Vod Gallery correctly 1`] = `
Array [
  <a
    href="https://www.twitch.tv/xqcow"
    onClick={[Function]}
    onMouseEnter={[Function]}
    rel="noopener noreferrer"
    target="_blank"
  >
    <div
      className="styles__StreamerInformation-sc-1grrpo5-0 jNzjxa"
    >
      <div
        title="Overwatch Professional tank player and full time streamer. EZ Clap"
      >
        <img
          alt="xQcOW"
          src="https://static-cdn.jtvnw.net/jtv_user_pictures/xqcow-profile_image-9298dca608632101-150x150.jpeg"
        />
      </div>
      <div>
        <h1>
          xQcOW
        </h1>
        <p>
          Overwatch Professional tank player and full time streamer. EZ Clap
        </p>
      </div>
    </div>
  </a>,
  <div
    className="styles__Container-sc-1grrpo5-1 fzloJj"
  >
    <div
      title="[ NoPixel ] FROM 20K TO 100K OR BUST. BEST PLAYER WORLD OF ALL GAMES, BIG BIG."
    >
      <button
        onClick={[Function]}
        type="button"
      >
        <div>
          <figure
            className="styles__Image-sc-1grrpo5-2 ezDIWJ"
          >
            <span>
              17:30:43
            </span>
          </figure>
        </div>
        <strong>
          [ NoPixel ] FROM 20K TO 100K OR BUST. BEST PLAYER WORLD OF ALL GAMES, BIG BIG.
        </strong>
        <span>
          <p>
            Views: 
            3.2M
          </p>
          <p>
            3/28/2021
          </p>
        </span>
      </button>
    </div>
    <div
      title="[ NoPixel ] BIG DAY TODAY OMG! COURT CASE, CAZINO JUMP AND MORE WOW! WARLORD MOMENT INBOUND"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        <div>
          <figure
            className="styles__Image-sc-1grrpo5-2 elKh"
          >
            <span>
              15:08:36
            </span>
          </figure>
        </div>
        <strong>
          [ NoPixel ] BIG DAY TODAY OMG! COURT CASE, CAZINO JUMP AND MORE WOW! WARLORD MOMENT INBOUND
        </strong>
        <span>
          <p>
            Views: 
            2.8M
          </p>
          <p>
            3/26/2021
          </p>
        </span>
      </button>
    </div>
  </div>,
]
`;
